src/backend/joanie/core/api/client.py

```diff
@@ -75,10 +75,52 @@ def get_queryset(self):
             if self.request.auth
             else self.request.user.username
         )
+        queryset = models.CourseProductRelation.objects
+        organization_id = self.kwargs.get("organization_id", None)
+        if organization_id:
+            queryset = queryset.filter(
+                organizations__id=organization_id,
+                organizations__accesses__user__username=username,
+            )
+        else:
+            queryset = queryset.filter(course__accesses__user__username=username)
+        return queryset.select_related("course", "product").prefetch_related(
+            "course__course_runs",
+            "course__organizations",
+            "course__products",
+            "organizations",
+            "product__target_courses",
+        )
 
-        return models.CourseProductRelation.objects.filter(
-            course__accesses__user__username=username
-        ).select_related("course", "product")
+
+class CoursePerOrganizationViewSet(
+    mixins.ListModelMixin, mixins.RetrieveModelMixin, viewsets.GenericViewSet
+):
+    """
+    API viewset for interactions with courses from an organization
+    """
+
+    lookup_field = "id"
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = serializers.CourseSerializer
+    ordering = ["-created_on"]
+    filterset_class = filters.CourseViewSetFilter
+
+    def get_queryset(self):
+        """
+        Custom queryset to limit to courses related to the organization's
+        """
+        username = (
+            self.request.auth["username"]
+            if self.request.auth
+            else self.request.user.username
+        )
+        organization_id = self.kwargs.get("organization_id", None)
+        queryset = models.Course.objects.filter(
+            organizations__id=organization_id,
+            organizations__accesses__user__username=username,
+        ).prefetch_related("organizations", "products", "course_runs")
+        return queryset
```

Comment:
I wonder if we could not reuse the existing CourseViewSet and just check if there is an `organization_id` provided within `kwargs` 🤔